packages/did/src/DidResolver/DidResolverV2.ts

```diff
@@ -0,0 +1,310 @@
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  resolutionOptions: DidResolverV2.ResolutionOptions = {}
+): Promise<DidResolverV2.ResolutionResult> {
+  try {
+    validateUri(did, 'Did')
+  } catch (error) {
+    return {
+      didResolutionMetadata: {
+        error: 'invalidDid',
+      },
+      didDocumentMetadata: {},
+    }
+  }
+
+  const resolutionResult = await resolveInternal(did)
+  if (!resolutionResult) {
+    return {
+      didResolutionMetadata: {
+        error: 'notFound',
+      },
+      didDocumentMetadata: {},
+    }
+  }
+
+  const { documentMetadata: didDocumentMetadata, document: didDocument } =
+    resolutionResult
+
+  return {
+    didResolutionMetadata: {},
+    didDocumentMetadata,
+    didDocument,
+  }
+}
+
+/**
+ * Implementation of `resolveRepresentation` compliant with W3C DID specifications (https://www.w3.org/TR/did-core/#did-resolution).
+ * Additionally, this function returns an id-only DID document in the case where a DID has been deleted or upgraded.
+ * If a DID is invalid or has not been registered, this is indicated by the `error` property on the `didResolutionMetadata`.
+ *
+ * @param did The DID to resolve.
+ * @param resolutionOptions The resolution options accepted by the `resolveRepresentation` function as specified in the W3C DID specifications (https://www.w3.org/TR/did-core/#did-resolution).
+ * @param resolutionOptions.accept The content type accepted by the requesting client.
+ * @returns The resolution result for the `resolveRepresentation` function as specified in the W3C DID specifications (https://www.w3.org/TR/did-core/#did-resolution).
+ */
+export async function resolveRepresentation(
+  did: DidDocumentV2.DidUri,
+  { accept }: DidResolverV2.DereferenceOptions<SupportedContentType> = {
+    accept: DID_JSON,
+  }
+): Promise<DidResolverV2.RepresentationResolutionResult<SupportedContentType>> {
+  if (!isValidContentType(accept)) {
+    return {
+      didResolutionMetadata: {
+        error: 'representationNotSupported',
+      },
+      didDocumentMetadata: {},
+    }
+  }
+
+  const { didDocumentMetadata, didResolutionMetadata, didDocument } =
+    await resolve(did)
+
+  if (didDocument === undefined) {
+    return {
+      // Metadata is the same, since the `representationNotSupported` is already accounted for above.
+      didResolutionMetadata,
+      didDocumentMetadata,
+    } as DidResolverV2.RepresentationResolutionResult<SupportedContentType>
+  }
+
+  const bufferInput = (() => {
+    if (accept === 'application/did+json') {
+      return JSON.stringify(didDocument)
+    }
+    if (accept === 'application/did+ld+json') {
+      const jsonLdDoc: DidDocumentV2.JsonLd<DidDocumentV2.DidDocument> = {
+        ...didDocument,
+        '@context': [W3C_DID_CONTEXT_URL, KILT_DID_CONTEXT_URL],
+      }
+      return JSON.stringify(jsonLdDoc)
+    }
+    // contentType === 'application/did+cbor
+    return cbor.encode(didDocument)
+  })()
```

Comment:
Idea for an optimisation:
```suggestion
  const encodeDocument = (() => {
    switch(accept) {
    case: 'application/did+json':
     return (didDocument) => JSON.stringify(didDocument)
    }
   case  'application/did+ld+json': 
   return (didDocument) =>{
      const jsonLdDoc: DidDocumentV2.JsonLd<DidDocumentV2.DidDocument> = {
        ...didDocument,
        '@context': [W3C_DID_CONTEXT_URL, KILT_DID_CONTEXT_URL],
      }
      return JSON.stringify(jsonLdDoc)
      }
  case 'application/did+cbor':
    return (didDocument) => cbor.encode(didDocument)
  default:
    return;
  })()
```
Move this up to the beginning of the function to replace isValidContentType; return 'representationNotSupported' if this does not return a function. If it does, you can call that function below to encode the did document

Reply:
Nice suggestion. I applied it here: https://github.com/KILTprotocol/sdk-js/pull/799/commits/79a75c1c45f69673cf3f8aca60f549c629bf0d16